### PR TITLE
Port diagnostic for certs with ephemeral keys on Windows to S.N.Q

### LINKED ISSUE #92721: [QUIC] Port ephemeral key diagnostics on Windows from SslStream
We should port https://github.com/dotnet/runtime/pull/83436 to Quic to get better diagnostic.

Original comment: https://github.com/dotnet/runtime/issues/92669#issuecomment-1737697195

### PR DESCRIPTION
**Pull request by rzikm**
Fixes #92721.

This ports logic from #83436 to System.Net.Quic.

Main changes are in MsQuicConfiguration and System.Net.Quic.csproj.

While implementing, I noticed that there is duplicated logic in TestHelper.cs and Configuration.Certificates.Dynamic.cs so I got rid of the one in TestHelper.

### CODE REVIEW

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.cs [hunk 1]
```diff
@@ -238,6 +239,15 @@ private static unsafe MsQuicSafeHandle Create(QuicConnectionOptions options, QUI
             {
                 ThrowHelper.ThrowIfMsQuicError(status, SR.net_quic_tls_version_notsupported);
             }
+
+            if (status == MsQuic.QUIC_STATUS_CERT_NO_CERT && certificate != null && certificate.HasPrivateKey())
+            {
+                using Microsoft.Win32.SafeHandles.SafeCertContextHandle safeCertContextHandle = Interop.Crypt32.CertDuplicateCertificateContext(certificate.Handle);
```
Comment (wfurt) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.cs:245]: Why do we need to duplicate?

**Review by wfurt — COMMENTED** [review]
looks good to me in general.

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.cs (hunk 1 above)
Comment (rzikm) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.cs:245]: This is the same logic as in SslStream. I assume the reason for that is that the Interop stub for `CertGetCertificateContextProperty` takes `SafeCertContextHandle`, but the `X509Certificate2.Handle` property returns an `IntPtr`. This is probably the easiest path to call that function, given that it's on slow/exceptional path anyway.

**Review by wfurt — APPROVED** [review]

**Comment by rzikm** [comment]
CI failures are known build errors.

**Pull request merged**